src/realm/sync/noinst/client_history_impl.hpp

```diff
@@ -246,13 +246,17 @@ class ClientHistory final : public _impl::History, public TransformHistory {
     /// about byte-level progress, this function updates the persistent record
     /// of the estimate of the number of remaining bytes to be downloaded.
     ///
+    /// \param transact If specified, it is a write transaction to be used to
+    /// commit the server changesets after they were transformed.
+    /// Note: Left in reading state when all tranformed changesets are commited.
+    ///
     /// \param transact_reporter An optional callback which will be called with the
     /// version immediately processing the sync transaction and that of the sync
     /// transaction.
     void integrate_server_changesets(
         const SyncProgress& progress, const std::uint_fast64_t* downloadable_bytes,
         util::Span<const RemoteChangeset> changesets, VersionInfo& new_version, DownloadBatchState download_type,
-        util::Logger&,
+        util::Logger&, TransactionRef transact = nullptr,
```

Comment:
do we want to take `transact` by const-ref here? also, do we ever call this without passing in an explicit value for `transact`? If not, let's not add a default value of nullptr.

Reply:
We call it without an explicit value for PBS and FLX state store messages. I changed it so there is always a const-ref provided.